android-example/build.rs

```diff
@@ -1,62 +1,107 @@
-    let jni_h = search_file_in_directory(&include_dirs, "jni.h").expect("Can not find jni.h");
-    let bitmap_h = search_file_in_directory(&include_dirs, "android/bitmap.h")
-        .expect("Can not find android/bitmap.h");
+        get_gcc_system_include_dirs(&target).expect("Can't get NDK's system include dirs");
 
+    let include_headers: Vec<_> = INCLUDE_SYS_H.into_iter()
+        .map(|h| search_file_in_directory(&include_dirs, h)
+             .expect(format!("Could not find header {}", h).as_ref()))
+        .collect();
+
+    let src_dir = Path::new(RUST_SRC_DIR);
     let out_dir = env::var("OUT_DIR").unwrap();
 
     gen_binding(
         &target,
         &include_dirs,
-        &[bitmap_h, jni_h],
+        &include_headers,
         &Path::new(&out_dir).join("android_c_headers.rs"),
     ).unwrap();
 
-    rust_swig_expand(
-        Path::new("src/java_glue.rs.in"),
-        &Path::new(&out_dir).join("java_glue.rs"),
-    ).unwrap();
-    println!("cargo:rerun-if-changed=src");
-    //if for some reason generated files were deleted, regenerate them
-    println!("cargo:rerun-if-changed={}", out_dir);
+    // Find files ending in .rs.in and expand them with SWIG
+    for _entry in WalkDir::new(RUST_SRC_DIR) {
+        let entry = _entry.expect("Error walking sources.");
+        if entry.path().is_dir() || !entry.path().to_string_lossy().ends_with(".rs.in") {
+            continue;
+        }
+
+        println!("Found SWIG specification: {}", entry.path().display());
+        let swigf = entry.path().strip_prefix("src").unwrap();
+        
+        rust_swig_expand(
+            &src_dir,
+            &swigf,
+            Path::new(&out_dir),
+        ).unwrap();
 
-    let cur_dir = env::current_dir().unwrap();
-    let arm_libs_path = cur_dir.join("src/obj/local/armeabi");
-    println!(
-        "cargo:rustc-link-search=native={}",
-        arm_libs_path.to_str().unwrap()
-    );
-    println!("cargo:rustc-link-lib=jnigraphics");
-    println!("cargo:rustc-link-lib=gnustl_shared");
-    println!("cargo:rustc-link-lib=atomic");
+        write_include_file(&src_dir, swigf).expect("Failed to write include file.");
+    }
+
+    // Hook up cargo reruns
+    println!("cargo:rerun-if-changed={}", RUST_SRC_DIR);
+    for dir in &include_dirs {println!("cargo:rerun-if-changed={}", dir.display());}
+    //if the generated files were deleted (e.g by gradle -q clean), regenerate them
+    println!("cargo:rerun-if-changed={}", out_dir);
 }
 
 fn get_gcc_system_include_dirs(target: &str) -> Result<Vec<PathBuf>, String> {
-    let gcc_cmd = if target == "i686-linux-android" {
-        "i686-linux-android-gcc"
-    } else {
-        "arm-linux-androideabi-gcc"
+    let gcc_cmd = match env::var("RUSTC_LINKER") {
+        Ok(path) => path,
+        Err(_) => target.to_owned() + "-gcc"
     };
 
-    let gcc_process = Command::new(gcc_cmd)
+    let gcc_path = Path::new(&gcc_cmd);
+    assert!(gcc_path.exists() && gcc_path.is_file(), format!(
```

Comment:
This `assert!` always failed for no `RUSTC_LINKER` case because of "arm-linux-androideabi-gcc" not absolute path